tests/test_position_sequence.py

```diff
@@ -0,0 +1,147 @@
+import pytest
+
+from useq import GridRelative, MDASequence
+
+
+@pytest.fixture
+def mda() -> MDASequence:
+    return MDASequence(
+        axis_order="tpgcz",
+        stage_positions=[(10, 20, 0), {"name": "test", "x": 10, "y": 20, "z": 50}],
+        channels=[{"config": "Cy5", "exposure": 50}],
+    )
+
+
+def _update_pos_sequence(mda: MDASequence, axis: str):
+    p_seq = MDASequence()
+    if "c" in axis:
+        p_seq = p_seq.replace(channels=[{"config": "DAPI", "exposure": 100}])
+    if "z" in axis:
+        p_seq = p_seq.replace(z_plan={"top": 60, "bottom": 55, "step": 1})
+    if "g" in axis:
+        p_seq = p_seq.replace(grid_plan={"rows": 2, "columns": 2})
+    if "t" in axis:
+        p_seq = p_seq.replace(time_plan=[{"interval": 1, "loops": 2}])
+    new_pos = [
+        (10, 20, 0),
+        {"name": "test", "x": 10, "y": 20, "z": 50, "sequence": p_seq},
+    ]
+    return mda.replace(stage_positions=new_pos)
+
+
+def event_list(sequence: MDASequence) -> list:
```

Comment:
i also think this makes the tests hard to read... you have to remember what order you picked up here (global index, then index, then pos_name) etc... Making the tests themselves very hard to look at